atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/futures/AtlasFutures.java

```diff
@@ -92,4 +96,18 @@ public void close() {
             throw Throwables.rewrapAndThrowUncheckedException(e);
         }
     }
+
+    /**
+     * Constructs a {@link Callable} which will be traced such that the trace id of the current trace is maintained when
+     * the wrapped {@link Callable} is passed to an {@link java.util.concurrent.Executor}.
+     *
+     * @param callable to be traced
+     * @param operation name of the passed callable
+     * @param <R> result type of he passed {@link Callable}
+     * @return traced {@link Callable}
+     */
+    public static <R> Callable<R> tracedCallable(Callable<R> callable, String operation) {
```

Comment:
Does this need to be public? Unclear where else it's used. On a separate note, will you not need to call it at points in three atlas stack where you've put things on executors as opposed to just the internal ski resort product?

Reply:
Yeah, should be private

Reply:
And also, just Tracers.wrap

Reply:
Removed.